src/globals.ts

```diff
@@ -0,0 +1,63 @@
+import chokidar from 'chokidar';
+import chalk from 'chalk';
+import Command from '@oclif/command';
+import { Specification } from './models/SpecificationFile';
+
+const GreenLog = chalk.hex('#00FF00');
+const OrangeLog = chalk.hex('#FFA500');
+const CHOKIDAR_CONFIG = {
+  // awaitWriteFinish: true // Used for large size specification files.
+
+};
+const WATCH_MESSAGES = {
+  logOnStart: (filePath: string) => console.log(GreenLog(`Watching AsyncAPI file at ${filePath}\n`)),
+  logOnChange: (handlerName: string) => console.log(OrangeLog(`Change detected, running ${handlerName}\n`)),
+  logOnAutoDisable: (docVersion: 'old' | 'new' | '' = '') => console.log(OrangeLog(`Watch mode for ${docVersion || 'AsyncAPI'} file was not enabled.`), OrangeLog('\nINFO: Watch works only with files from local file system\n'))
+};
+
+type CHOKIDAR_INSTANCE_STORE = Record<string, boolean>
+
+const IS_CHOKIDAR_INSTANCE_RUNNING: CHOKIDAR_INSTANCE_STORE = {
+  _default: false,
+};
+
+export type specWatcherParams = {
+  spec: Specification,
+  handler: Command,
+  handlerName: string,
+  label?: string,
+  docVersion?: 'old' | 'new';
+}
+
+const getLabelValue = (label: string): boolean => {
+  return IS_CHOKIDAR_INSTANCE_RUNNING[label as string];
+};
+const setLabelValue = (label: string, val: boolean) => {
+  IS_CHOKIDAR_INSTANCE_RUNNING[label as string] = val;
+};
+
+export const specWatcher = (params: specWatcherParams) => {
+  if (!params.spec.getFilePath()) { return WATCH_MESSAGES.logOnAutoDisable(params.docVersion); }
+  if (!params.label && getLabelValue('_default')) { return; }
+  if (params.label && getLabelValue(params.label as string)) { return; }
```

Comment:
Why do you want to use normal object, not `Map`? I think that it's better here. Really, please consider to use `Map`, Then you won't need to write that `getLabelValue`, `setLabelValue` functions because you have `.set`, `.get` in the `Map` prototype.

Also, if you don't have `params.label` you don't need to write separate condition for `_default` label but you can also do:

```ts
const hasLabel = getLabelValue(params.label || '_default');
// and
setLabelValue(params.label || '_default', true);
```

Reply:
Sure please review the latest changes :)